xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -594,4 +609,15 @@ public void displayView(StringBuffer buffer, String name, String prefix, BaseCol
             buffer.append(getDisplayValue(prop.getValue(), name, map, context));
         }
     }
+
+    private String getPropertyType(String className, String propertyName, XWikiContext context) throws Exception
+    {
+        PropertyClass pc = null;
+        try {
+            pc = (PropertyClass) context.getWiki().getDocument(className, context).getXClass().get(propertyName);
+        } catch (XWikiException e) {
+            LOGGER.error("Failed to get property [{}] for document [{}]", propertyName, className, e);
+        }
+        return pc.newProperty().getClass().getName();
```

Comment:
@zoubir you didn't fix or prevent the ``NullPointerException`` that you can have in ``getPropertyType()``. In your last commit you've added a try/catch to catch the ``NullPointerException`` from ``getPropertyType()``  as if it is an expected exception. You should prevent the ``NullPointerException`` not catch it.